### PR TITLE
Scc 2591

### PR DESCRIPTION
**Pull request by danamansana**
- Adds a new method to reconstruct the original url
- Adds the `originalUrl` param in the new (redirected) url.
- Fixes the tests to expect this parameter
- Updates the sample events to include `host`

### CODE REVIEW

**Review by nonword — COMMENTED** [review]
Looks good. Just some light suggestions. Also, since we're using `multiValueQueryStringParameters`, could we test the case where a param has multiple values?

File: index.js [hunk 1]
```diff
@@ -74,11 +74,19 @@ const expressions = {
   },
 };
 
+function reconstructOriginalURL(path, query, host, method) {
+  const reconstructedQuery = Object.entries(query).map(([key, values]) => {
```
[the remark this answers is not in the document]
Comment (danamansana) [reply index.js:78]: I've added a default value of `{}` for `query`.

File: index.js [hunk 2]
```diff
@@ -74,11 +74,19 @@ const expressions = {
...
+      return values.map(value => value.length ? `${key}=${value}` : key).join('&')
```
[the remark this answers is not in the document]
Comment (danamansana) [reply index.js:79]: This is just to try to match Webpac as closely as possible when we display the original url to the patron (Webpac has query params missing `=`).

File: index.js [hunk 3]
```diff
@@ -97,7 +106,8 @@ const handler = async (event, context, callback) => {
     let path = event.path;
     let query = event.multiValueQueryStringParameters;
     let method = event.multiValueHeaders['x-forwarded-proto'][0] ;
```
[the remark this answers is not in the document]
Comment (danamansana) [reply index.js]: Fixed.

**Comment by danamansana** [comment]
Any advice on testing the `multiValueQueryStringParameters`. I don't think we actually know of any cases where there are mutliple values for the same param, it's just that the lambda is set up to receive `multiValueQueryStringParameters` in the event

**Comment by nonword** [comment]
Re: testing `multiValueQueryStringParameters`, if we have no known cases where params are repeated, then I'm less interested in testing. If there are any legacy request paths that use a param with multiple values, our tests should define how we handle the different values. But if there are no known uses, then whatever translation would currently happen in that unlikely scenario is probably sufficient (assuming it doesn't raise an error).

**Review by nonword — APPROVED** [review]

**Pull request merged**